src/Controls/src/Core/ObservableWrapper.cs

```diff
@@ -291,10 +298,7 @@ public void Remove(object value)
 			Remove((TRestrict)value);
 		}
 
-		public void CopyTo(Array array, int index)
-		{
-			CopyTo(array.Cast<TRestrict>().ToArray(), index);
-		}
+		public void CopyTo(Array array, int index) => _list.CopyTo((TTrack[])array, index);
```

Comment:
I think you need to call the CopyTo method on line 57. That way only the `TRestrict` items get copied into the array.